Added Support for TSQL's column_privileges View

oscarno22 opened the pull request:
### Description

Currently, Babelfish does not support
information_schema_tsql.constraint_table_usage. This commit implements
this view. It should also be noted that the function sys.schema_name was
moved from import_export_compatibility.sql to sys_function.sql. The
function was being called in the view definition before it was defined.
This updates adjusts for the order of file execution.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>
 
### Issues Resolved

- The view information_schema_tsql.constraint_table_usage not supported
- The function sys.schema_name being called before definition


### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

oscarno22 committed:
Added Support for TSQL's column_privileges View

Currently, Babelfish does not support
information_schema_tsql.constraint_table_usage. This commit implements
this view. It should also be noted that the function sys.schema_name was
moved from import_export_compatibility.sql to sys_function.sql. The
function was being called in the view definition before it was defined.
This updates adjusts for the order of file execution.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>

shalinilohia50 commented on contrib/babelfishpg_tsql/sql/information_schema_tsql.sql:
```diff
@@ -716,3 +716,23 @@ CREATE OR REPLACE VIEW information_schema_tsql.routines AS
+*/
+CREATE OR REPLACE VIEW information_schema_tsql.constraint_table_usage
+AS SELECT sys.db_name()::sys.nvarchar(128) AS table_catalog,
+    sys.schema_name(nr.oid)::sys.nvarchar(128) AS table_schema,
```
Column names should be in Block letters.

shalinilohia50 replied:
Use column "orig_name" of  `sys.babelfish_namespace_ext` to get the value for "TABLE_SCHEMA". We don't need to move the function "schema_name"

oscarno22 committed:
Updated Column Names to Block Letters

Currently, column names in the view definition are lowercase. This
commit updates them to block letters.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>

shalinilohia50 commented:
General comments for tests:
1. Create the test input files as `.sql` and output files as `.out`. 
2. We should split the tests in prepare, verify and cleanup format. Name the input files as `constraint_table_usage-vu-prepare.sql`, `constraint_table_usage-vu-verify.sql`, `constraint_table_usage-vu-cleanup.sql` and expected output files as  `constraint_table_usage-vu-prepare.out`, `constraint_table_usage-vu-verify.out`, `constraint_table_usage-vu-cleanup.out`

shalinilohia50 commented on test/JDBC/input/constraint_table_usage.txt:
```diff
@@ -0,0 +1,33 @@
+use master;
```
To execute any query in JDBC, you will have to write `GO` after the statement.

oscarno22 committed:
Updated View Columns

This commit corrects the column names to the correct block letters
format. It also updates the "::" syntax to "CAST()" for corresponding
data types.

TASK:

Signed-off-by: Ozzy Nolen <oscarno@amazon.com>
